src/compiler/parse/state/tag.ts

```diff
@@ -291,6 +292,29 @@ function read_tag_name(parser: Parser) {
 	return name;
 }
 
+function node_to_template_literal(value: TemplateNode[]): TemplateLiteral {
+
+	const literal: TemplateLiteral = {
+		type: "TemplateLiteral",
+		expressions: [],
+		quasis: []
+	};
+
+	value.forEach((node) => {
+		if (node.type === "Text") {
+			literal.quasis.push({
+				type: 'TemplateElement',
+				value: { raw: node.raw, cooked: null },
+				tail: false
+			});
+		} else if (node.type === "MustacheTag") {
+			literal.expressions.push(node.expression);
+		}
+	});
+
+	return literal;
+}
+
```

Comment:
@tanhauhau I think I got it! How's this look?

Reply:
i fixed the case when there's no text before any expression:

```js
<div style:padding="{padding}px" />
```

you would need to insert an empty `quasi` into the `TemplateLiteral`